Sources/NIOCore/ByteBuffer-lengthPrefix.swift

```diff
@@ -149,4 +149,31 @@ extension ByteBuffer {
 
         return messageBuffer
     }
+
+    /// Reads an integer length prefix at the current `readerIndex`, then returns a `ByteBuffer` slice of that length
+    /// without advancing the reader index.
+    ///
+    /// This method is equivalent to calling `getLengthPrefixedSlice(at: readerIndex, ...)`.
+    ///
+    /// - Parameters:
+    ///   - endianness: The endianness of the length prefix (defaults to big endian).
+    ///   - integer: The `FixedWidthInteger` type for the length prefix.
+    /// - Returns: `nil` if the length prefix could not be read, the length prefix is negative, or if there aren't enough
+    ///            bytes for the message after the prefix. Otherwise, a slice of the requested length.
+    @inlinable
+    public func peekLengthPrefixedSlice<Integer>(
+        endianness: Endianness = .big,
+        as integer: Integer.Type
+    ) -> ByteBuffer? where Integer: FixedWidthInteger {
+        guard let lengthPrefix = self.getInteger(at: self.readerIndex, endianness: endianness, as: Integer.self),
+            let messageLength = Int(exactly: lengthPrefix),
+            let messageBuffer = self.getSlice(
+                at: self.readerIndex + MemoryLayout<Integer>.size,
+                length: messageLength
+            )
+        else {
+            return nil
+        }
+        return messageBuffer
```

Comment:
Can't we just call `getLengthPrefixedSlice` here?

Reply:
Yeah I thought about that, but I went with this approach just because of how `getLengthPrefixedSlice` is implemented, it's doing multiple things. It reads an integer prefix, then reads a slice relative to that prefix, fails on some inputs (negative or out-of-bounds), etc. So it didnt seem as safe as say implementing a peek variant of getUUID which is predictable and doesn't have side-effects, so I could just create a wrapper for it.

If however it's more reasonable to simply call `getLengthPrefixedSlice`, I can change it to that. I can see how the current version is just duplicate code that needs to be separately maintained (unless ofcourse get*s will eventually be phased out anyway!)

Reply:
I think this should probably just call `getLengthPrefixedSlice`.

Reply:
Cory beat me to it but yes I think it's better to just call `getLengthPrefixedSlice`. Copying the code doesn't really make e.g. the input validation any more obvious; and it makes maintainability potentially harder as we now have the same code in multiple places as you rightly pointed out.